frontend/datagathering.cpp

```diff
@@ -96,12 +107,16 @@ void MainWindow::handle_result_cycles(HttpRequestWorker *worker)
             viewMenu_->addAction(checkableAction);
             connect(checkBox, SIGNAL(stateChanged(int)), this, SLOT(columnHider(int)));
 
-            desiredHeader_ = getFields(ui_->instrumentsBox->currentText(), ui_->instrumentsBox->currentData().toString());
-
             // Filter table based on desired headers
-            if (desiredHeader_.contains(key))
-                checkBox->setCheckState(Qt::Checked);
-            else
+            auto found = false;
+            foreach (const auto &header, desiredHeader_)
+                if (std::get<0>(header) == key)
+                {
+                    checkBox->setCheckState(Qt::Checked);
+                    found = true;
+                    break;
+                }
+            if (!found)
```

Comment:
Another possible use of `std::find_if` or possibly `std::any_of` might be even simpler.